--- a/src/qtdeclarative.rs
+++ b/src/qtdeclarative.rs
@@ -28,7 +28,7 @@
     #include <memory>
     #include <QtQuick/QtQuick>
     #include <QtCore/QDebug>
-    #include <QtWidgets/QApplication>
+    #include <QtGui/QGuiApplication>
     #include <QtQml/QQmlComponent>
 
     struct SingleApplicationGuard {
@@ -47,12 +47,12 @@
     };
 
     struct QmlEngineHolder : SingleApplicationGuard {
-        std::unique_ptr<QApplication> app;
+        std::unique_ptr<QGuiApplication> app;
         std::unique_ptr<QQmlApplicationEngine> engine;
         std::unique_ptr<QQuickView> view;
 
         QmlEngineHolder(int &argc, char **argv)
-            : app(new QApplication(argc, argv))
+            : app(new QGuiApplication(argc, argv))
             , engine(new QQmlApplicationEngine())
         {}
     };
